### PR TITLE
[DOC]Install Pachctl Autocompletion

### LINKED ISSUE #4495: Include terminal completion setup process in installation docs
Currently, the [installation instructions](https://docs.pachyderm.com/latest/getting_started/local_installation/) don't include instructions for installing terminal completions. These instructions are necessary because [it's not possible to automatically install the completions.](https://github.com/pachyderm/pachyderm/issues/2371)

See the [kubectl installation docs](https://kubernetes.io/docs/tasks/tools/install-kubectl/#optional-kubectl-configurations) for an example docs on how to install completion and ensure the completions are functional in bash/zsh. Our installation instructions would be almost identical, except we offer some utility functionality to help install the completions, via the `--install` and `--path` flags for `pachctl completion bash` and `pachctl completion zsh`.

Blocked by #4494, which adds zsh completion and changes the completion command line syntax.

### PR DESCRIPTION
**Pull request by svekars**
Closes #4495

### CODE REVIEW

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+1. Verify that `bash` or `zsh` completion is installed on your machine.
+   For example, if you have installed bash completion by using Homebrew,
+   type:
+
+   ```bash
+   brew info bash-completion
+   ```
+
+   This command returns information about the directory in which
+   `bash-completion` is installed.
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: the `bash-completion` package and bash completion scripts are installed in separate locations. `brew info bash-completion` will return info about both paths.

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Pachyderm autocompletion is supported for both `bash` and `zsh`.
+You must have either of them preinstalled
+before installing Pachyderm autocompletion.
+
+!!! tip
+    Type `pachctl completion --help` to display help information about
+    the command.
+
+To install `pachctl` autocompletion, perform the following steps:
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: these following instructions are for installing bash completions. do we want to have info about how to install zsh completions as well?

Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: added the zsh instructions

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+   ```bash
+   Bash completions installed in /usr/local/etc/bash_completion.d/pachctl, you must restart bash to enable completions.
+   ```
+
+   * If you are using `zsh`, run the following command:
+
+     ```sh
+     pachctl completion zsh --install
+     ```
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: You need to specify the path for zsh as well

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+   the path to `bash_completion.d`. Also, the output of the info
+   command, might have a suggestion to include the path to
+   `bash-completion` into your `~/.bash_profile` file.
+
+1. Install `pachctl` autocompletion:
+
+   * If you are using `bash`, run the following command:
+
+     ```bash
+     pachctl completion bash --install --path <path-to-bash>
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: It's not the path to bash, it's the path to the bash completion scripts

**Comment by ysimonson** [comment]
I wonder if it would be more straight-forward to entirely separate the bash and zsh installation instructions

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+   ```bash
+   brew info bash-completion
+   ```
+
+   This command returns information about the directory in which
+   `bash-completion` and bash completion scripts are installed.
+   For example,  `/usr/local/etc/bash_completion.d/`. Unless it is
+   the default `/etc/bash_completion.d/` location, you need to specify
+   the path to `bash_completion.d`. Also, the output of the info
+   command, might have a suggestion to include the path to
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]:
```suggestion
   command might have a suggestion to include the path to
```

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+   type:
+
+   ```bash
+   brew info bash-completion
+   ```
+
+   This command returns information about the directory in which
+   `bash-completion` and bash completion scripts are installed.
+   For example,  `/usr/local/etc/bash_completion.d/`. Unless it is
+   the default `/etc/bash_completion.d/` location, you need to specify
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: `/etc/bash_completion.d/` is not the default for homebrew-installed `bash-completion`. Technically, bash-completion doesn't have a default, although `/etc/bash_completion.d/` is commonly employed. Given that, I'd just strike out this sentence, or reword it.

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+   pachctl completion bash --install --path <path/to/bash-completion>
+   ```
+
+   For example, if you specify the path to `bash-completion` as
+   `/usr/local/etc/bash_completion.d/pachctl`, your system response
+   looks like this:
+
+   **System response:**
+
+   ```bash
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: this should not be bash highlighted

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+Recently, `zsh` became the default shell on macOS, therefore, many users
+might prefer using `zsh`. Before you install pachctl completion for zsh,
+you must have `zsh-completions` installed.
+
+To install `pachctl` completion for `zsh`, complete the following
+steps:
+
+1. Verify that `zsh-completions` are installed on your machine:
+
+   ```bash
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: this should be zsh highlighted

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+   brew info zsh-completions
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: this instruction is specific to homebrew - we should have a caveat about that, like in the bash instructions

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+   ```
+
+   You should see the directory in which `zsh-completions` are installed
+   and instructions to add the correct path in the ~/.zshrc file. Make sure
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]:
```suggestion
   and instructions to add the correct path in the `~/.zshrc` file. Make sure
```

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+   **Example:**
+
+   ```sh
+   pachctl completion zsh --install --path /usr/local/share/zsh-completions/_pachctl
+   ```
+
+   **System response:**
+
+   ```sh
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]: this should not be highlighted

File: doc/docs/master/getting_started/install-pachctl-completion.md (hunk 6 above)
Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: reworded

File: doc/docs/master/getting_started/install-pachctl-completion.md (hunk 5 above)
Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: fixed

File: doc/docs/master/getting_started/install-pachctl-completion.md (hunk 7 above)
Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: fixed.

File: doc/docs/master/getting_started/install-pachctl-completion.md (hunk 8 above)
Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: I'm 100% sure it's supported but let's try

File: doc/docs/master/getting_started/install-pachctl-completion.md (hunk 9 above)
Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: added

File: doc/docs/master/getting_started/install-pachctl-completion.md (hunk 10 above)
Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: why? We typically use code highlighting for names of files.

File: doc/docs/master/getting_started/install-pachctl-completion.md (hunk 11 above)
Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: fixed.

File: doc/docs/master/getting_started/install-pachctl-completion.md (hunk 10 above)
Comment (ysimonson) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: on the following line, the path is backticked

Comment (svekars) [reply doc/docs/master/getting_started/install-pachctl-completion.md]: yes, right. My bad!

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+   ```
+
+1. Restart your terminal.
+
+   `pachctl` autocomplete should now be enabled in your system.
+
+## Install pachctl Autocompletion for `zsh`
+
+Recently, `zsh` became the default shell on macOS, therefore, many users
+might prefer using `zsh`. Before you install pachctl completion for zsh,
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]:
```suggestion
might prefer using `zsh`. Before you install `pachctl` completion for `zsh`,
```

File: doc/docs/master/getting_started/install-pachctl-completion.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+you must have `zsh-completions` installed.
+
+To install `pachctl` completion for `zsh`, complete the following
+steps:
+
+1. Verify that `zsh-completions` are installed on your machine.
+   For example, if you have installed bash completion by using Homebrew,
```
Comment (ysimonson) [inline doc/docs/master/getting_started/install-pachctl-completion.md]:
```suggestion
   For example, if you have installed zsh completion by using Homebrew,
```

**Comment by svekars** [comment]
This should go to 1.10 and should not be merged before that.

**Pull request merged**